### ISSUE TITLE
Random freezes on files drag 'n drop

### ISSUE THREAD

**Issue by jesobreira** [elementary/files#890]
Sometimes when I try dragging and dropping files, the cursor shows a pointer with a "plus" sign. When it happens, the system stops responding to clicks, although I am able to use a few (but not all) keyboard shortcuts. I am unable to use Super + S or Alt + Tab, but I can use Ctrl + Alt + F1/F3/F7 and even Super + L when it happens.

I am able to move the mouse, only the clicks won't get recognized.

It happens often (at least once a day), but not every time I try dragging and dropping files (most of the times it works well and this issue does not happen). I am only able to get out from the stuck state by resetting my machine.

Environment information: Elementary OS Loki, 8 GB RAM, using standard (2 button + wheel) mouse (not a trackpad).

Steps to reproduce:

1. Open Files
2. Click and hold the click on any file
3. Move the mouse

Expected result: the file gets moved to where the mouse click is unheld

Got result: most of the times it works well, but sometimes, the mouse cursor becomes as in the following picture and the machine stops responding to clicks.

<a href="https://ibb.co/xFC6m0g"><img src="https://i.ibb.co/PDmgTnW/FDF2-A36-A-2-AF4-49-AF-BF41-23-B2-D8713805.jpg" alt="FDF2-A36-A-2-AF4-49-AF-BF41-23-B2-D8713805" border="0" /></a>

**Comment by jesobreira** [comment]
I continue having this issue at least once a day. It's so repeating that I've placed a sticky note on my monitor to make me remember the Super + Shift + RSEIUB kernel reset command. Also, something interesting I noted was that when this issue happens and I try using any of the shell interfaces (Ctrl + Alt + F1/F2/F3), it requires me to login. I put my username with no problem, but when it asks for my password, it auto-detects some return key (that I obviously didn't press) instantly, making it impossible for me to insert my password.

The keyboard is new and has no problem.

I'll be returning to Ubuntu since it's impossible to work under that <s>risk</s> certainty of having to restart during important tasks, but I will try to keep this harddisk with elementary OS installed for a few days, so if some testing is needed, I will still be able to perform it.

**Comment by yuliantosb** [comment]
I got the same issue here, any solution?

**Comment by bleakgrey** [comment]
I can confirm this is happening to Juno as well

**Comment by brcmesquita** [comment]
I got the same issue here, and I think they haven't a solution yet.
But, I've discovered, when I trying ALT+F4 (a old shortcut to CLOSE in Windows), the Pantheon Files is closed and I gain control of eOS again. Then, the plus signal disapear. I'm using elementary OS 5 Juno.

**Comment by brcmesquita** [comment]
I have another issue when I'm using drag and drop. Sometimes the drag-and-drop functionality stops working by itself and does not come back. You need to close the Files and open again for the drag and drop function to work again.

**Comment by rochabianca** [comment]
I have the same issue, It happens every time I grab something and try to drop it on VS Code, I call it "the grab hell"

**Comment by Julioevm** [comment]
This happens to me if I try to drag 2 files into telegram web app, if I just drag one it might happen but not as often.

**Comment by jeremypw** [comment]
There is an outstanding PR related to DnD issues (#814).  It is not clear whether this will fix this particular issue as this seems random and therefore not easily reproduced.  If anyone experiencing this issue could try out that PR it would be helpful.  DnD copies could also trigger the bug that the unmerged PR #714 fixes.

**Comment by citrusui** [comment]
I am not sure if this is related, but I have an almost identical issue when extracting files from an archive.

How to reproduce:

- Open Files
- Open an archive (zip, tar, etc)
- Drag and drop any file or file(s) from Archive Manager into the Files window
- Click and drag any file from the Files window

Upon completing the last step, the file or folder you have dragged will be duplicated. The cursor will be stuck on a "grabbing" animation with a small green plus in the corner, [as shown here.](https://ibb.co/xFC6m0g)

The only way to "fix" this is by forcibly closing Files, usually with Alt+F4. Very annoying bug that's been persisting for a few months now.

**Comment by jeremypw** [comment]
@citrusui I can confirm your version of this bug and also that it is fixed by PR #814.  Unfortunately no-one has yet come forward to review/approve this PR and unless and until someone does, it cannot be merged and released.

**Comment by jeremypw** [comment]
@jesobreira @citrusui Could you confirm whether you are still experiencing this with version 4.1.8 (or 4.1.9 if you have it). Thanks.

**Comment by citrusui** [comment]
@jeremypw Unfortunately, yes, I do still experience this problem. I'm running `pantheon-files` version 4.1.9+r3433+pkg82~ubuntu5.0.1 if that helps

**Comment by brcmesquita** [comment]
> @jesobreira @citrusui @brcmesquita @rochabianca Could you confirm whether you are still experiencing this with version 4.1.8 (or 4.1.9 if you have it). Thanks.

No. It hasn't happened to me anymore.

Here are some tips:
- Keep elementary up to date (sudo apt update -y && sudo apt upgrade -y);
- When this happens, do nothing else. Just hit CTRL + Q and see if it's solved;
- Drag files/folders with the right mouse button instead of the left. The difference is that you will have a menu with copy or move options when you finish dragging.

**Comment by jeremypw** [comment]
@citrusui @brcmesquita Thanks for the feedback! I'll try to fix this soon.

**Comment by Hedronmx** [comment]
I still have this problem, especially when dragging from zip file and then dragging from the main window

**Comment by Suzamax** [comment]
Can confirm, it happens half the time while moving files between tabs.

**Comment by jeremypw** [comment]
I had forgotten that #814 (which was supposed to fix this)  was never reviewed or merged and is now awaiting splitting into smaller PRs.  I'll try and get on with it in the near future unless another dev would like to ...

**Comment by Hedronmx** [comment]
@jeremypw  Please do!! I appreciate you work :smile:

**Comment by jeremypw** [comment]
@Hedronmx , @Suzamax  Please could you confirm whether this occurs during dragging or after dropping?  Does it make difference if the drag involves the "auto-open" function - i.e. hovering over a folder until it opens?

**Comment by Suzamax** [comment]
@jeremypw during dragging. If you stop holding the mouse click button, it keeps the dragging and you can't click to drop the file. Didn't try the auto-open folder thing. It's only between tabs in my case.

**Comment by Hedronmx** [comment]
@jeremypw during dragging as well, but on my end it happens after drag and dropping a file from a zip or any other archive manager file into the elementary file windows. 
After doing that, every drag I do inside the elementary file window glitches out and it keeps dragging unless I search a file inside the elementary file window by typing a name.

**Comment by jeremypw** [comment]
@Hedronmx , @Suzamax  Thanks for the further information.

**Comment by jeremypw** [comment]
I do not experience any problem drag/dropping files between Files and  a file-roller window, using the current master source (with default 'single-click' mode).  A couple of small DnD related changes were merged.

**Comment by jeremypw** [comment]
I have found a reliable method of breaking DnD:

* Using the secondary (right) button drag on the empty background. This starts a drag (in itself a bug).
* Drop the non-existent file onto the background. Nothing appears to happen
* Drag a real file using the secondary button.  The DnD will break. There will be no drag icon and releasing the button does not end the drag.

Hopefully with a reproducible issue, I can find a solution.

**Comment by jeremypw** [comment]
I think I will create a separate issue for the secondary drag problem mentioned above at is not clear it is exactly the same as that originally reported.

**Comment by Hedronmx** [comment]
Please test with archive manager. I get it every time I drag and drop from Archive to Files

**Comment by jeremypw** [comment]
File-roller is the default archive manager - which one are you using?  I tried with `.7z` and `.zip` archives.

**Comment by Hedronmx** [comment]
I am using Fileroller, and using left click. I suppose that's primary button.

Want me to record a video?

**Comment by jeremypw** [comment]
@Hedronmx Are you using Juno stable or do have the daily repos?  Let me try again with a unmodified stable installation. I normally use the unstable version.

**Comment by Hedronmx** [comment]
I am running stable version. Thank you for your time

**Comment by jeremypw** [comment]
@Hedronmx I did not have any problems dragging files out of an Archive Manager window into Files on Juno stable with an English (UK) locale.  Perhaps a video would help, I may be able to spot something.  Let me know if you have made any changes to any default settings and which locale you are using.

**Comment by Hedronmx** [comment]
Here it is.

https://gfycat.com/jovialgreenblackrussianterrier

I am using english us as locale and I have not made any significant changes to the default settings. Just adding another keyboard layout (spanish) and setting up night light

**Comment by jeremypw** [comment]
Ah!  I can reproduce now.  I thought the problem affecting dragging *any* file from an archive into Files. I was testing using a pre-existing archive.  If I follow the video and drag a file back out of an archive just created in the same view then I do get the freeze.

**Comment by jeremypw** [comment]
I'll create a separate issue as the archive problem is reproducible, not random.

**Comment by Hedronmx** [comment]
It happens with downloaded archives too! Not just with the newly created archives.

**Comment by jeremypw** [comment]
I have traced the problem to the Files handling of "XdndDirectSave0" targets, which some external apps use.  Internal Dnd in Files uses "text/uri" targets.  I have a solution but I am working on making it as simple as possible so that it gets reviewed/merged in a reasonable time.  Really the DnD code needs a complete refactor, which I did do, but could not get it reviewed/merged as a single PR.

**Comment by Hedronmx** [comment]
Awesome!! If you need me to test this out i am willing to help!

**Comment by jeremypw** [comment]
@Hedronmx Here is the PR https://github.com/elementary/files/pull/1087.  It needs testing to make sure there are no regressions with other DnD operations.  There are some manual tests here: https://github.com/elementary/files/wiki/dnd but this is not necessarily exhaustive.  If you are able to build and test this PR and comment on your findings in the PR, that would help to get it merged. Make sure you close any existing Files windows before running. Thanks.

**Comment by jeremypw** [comment]
Closing this as more specific reproducible issues need to be opened if still extant for Horus.
